base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Basedecorator
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
- [ ] I will encourage you to add the getter and setter so that the nameable file can be fully accessed.
- [ ] Also, remember the inheritance `< Nameable`